sys/vane/jael.hoon

```diff
@@ -1563,20 +1971,68 @@
     |=  rep=response:rpc:jstd
     ^+  +>
     ?<  ?=(%batch -.rep)
+    ?<  ?=(%fail -.rep)
     ?:  ?=(%error -.rep)
-      ?.  =('filter not found' message.rep)
-        ~&  [%unhandled-filter-error message.rep]
+      ?.  ?|  =('filter not found' message.rep)  ::  geth
+              =('Filter not found' message.rep)  ::  parity
+          ==
+        ~&  [%unhandled-filter-error +.rep]
         +>
-      ~&  %filter-timed-out--recreating
+      ~&  [%filter-timed-out--recreating block=latest-block +.rep]
       new-filter
     ::  kick polling timer, only if it hasn't already been.
     =?  +>  ?&  ?=(%| -.source)
                 (gth now poll-timer.p.source)
             ==
       wait-poll
-    ?>  ?=(%a -.res.rep)
+    (take-events rep)
+  ::
+  ::  +take-block-number: take block number and start catching up
+  ::
+  ++  take-block-number
+    |=  rep=response:rpc:jstd
+    ^+  +>
+    ?<  ?=(%batch -.rep)
+    ?<  ?=(%fail -.rep)
+    ?:  ?=(%error -.rep)
+      ~&  [%take-block-number-error--retrying message.rep]
+      get-latest-block
+    =.  foreign-block  (parse-eth-block-number res.rep)
+    ~&  [%setting-foreign-block foreign-block]
+    (catch-up latest-block)
+  ::
+  ::  +take-catch-up-step: process chunk
+  ::
+  ++  take-catch-up-step
+    |=  [rep=response:rpc:jstd from-block=@ud next-block=@ud]
+    ^+  +>
+    ?<  ?=(%batch -.rep)
+    ?<  ?=(%fail -.rep)
+    ?:  ?=(%error -.rep)
+      ~&  [%catch-up-step-error--retrying message.rep]
+      (catch-up from-block)
+    ::  XX file
```

Comment:
It's not immediately clear what this means, what should be done here.

Reply:
This is obsolete; will remove